### PR TITLE
feat(contrib): docker-push plugin

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-cds#19998.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19998.1 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19998.2 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by richardlt — APPROVED** [review]

**Review by yesnault — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#20006.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#20006.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

File: contrib/grpcplugins/action/docker-push/docker-push.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  image:
+    type: string
+    description: Image name
+    required: true
+  tags:
+    type: string
+    description: |- 
+      The tags to associate with the image on the registry.
+
+      This parameter can be empty if you want to keep the same tag.
```
Comment (sguiheux) [inline contrib/grpcplugins/action/docker-push/docker-push.yml:16]: You return an eror when the field is empty

File: contrib/grpcplugins/action/docker-push/main.go [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+	imageID    string
+	created    string
+	size       string
+}
+
+func (actPlugin *dockerPushPlugin) perform(ctx context.Context, image string, tags []string, registry, registryAuth string) error {
+	if image == "" {
+		return sdk.Errorf("wrong usage: <image> parameter should not be empty")
+	}
+	if len(tags) == 0 {
```
Comment (sguiheux) [inline contrib/grpcplugins/action/docker-push/main.go:80]: You say that tags can be empty in plugin descriptor file

File: contrib/grpcplugins/action/docker-push/main.go [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+				break
+			}
+		}
+		if !manifestFound {
+			return nil, time.Since(t0), errors.New("unable to get uploaded image manifest")
+		}
+
+	default:
+		// Push on the registry set as parameter
+		destination = registry + "/" + img.repository + ":" + tag
```
Comment (sguiheux) [inline contrib/grpcplugins/action/docker-push/main.go:254]: Maybe we have to failed if (no registry && no auth) ?

File: contrib/grpcplugins/action/docker-push/main.go [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+
+	response, err := grpcplugins.CreateRunResult(ctx, &actPlugin.Common, &runResultRequest)
+	if err != nil {
+		return nil, time.Since(t0), err
+	}
+
+	result := response.RunResult
+
+	var destination string
+	// Upload the file to an artifactory or CDN
```
Comment (sguiheux) [inline contrib/grpcplugins/action/docker-push/main.go:163]: We do not upload to CDN

**Review by sguiheux — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#20009.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#20009.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by yesnault — APPROVED** [review]

**Review by sguiheux — APPROVED** [review]

**Pull request merged**